components/autocomplete/index.js

```diff
@@ -231,13 +232,36 @@ export class Autocomplete extends Component {
 		}
 	}
 
-	loadOptions( index ) {
-		this.props.completers[ index ].getOptions().then( ( options ) => {
-			const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + i } ) );
+	loadOptions( completerIndex ) {
+		const completer = this.props.completers[ completerIndex ];
+		const { type } = completer;
+
+		// Filter the options getter itself because we only want to invoke one,
+		// whether it be default or custom.
+		const optionsGetter = applyFilters( `autocomplete.${ type }.options.getter`, completer.getOptions );
```

Comment:
We need to standardize the way we name filters to have one pattern that is easy to follow. So far we were prefixing the hooks name with the package name followed by the component's name. So, in this case it would be something like: `components.Autocomplete`. Another thing here is that we can combine those 3 filters into one as follows and call it once only when the component gets initialized:

```js
this.props.completers = applyFilters( `components.Autocomplete.completers', this.props.defaultCompleters );
```